server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -330,6 +330,15 @@ public long relativeTimeInNanos() {
         return cachedTimeThread.relativeTimeInNanos();
     }
 
+    /**
+     * Returns a value of nanoseconds that may be used for absolute time calculations.
+     *
+     * This method can be used for calculating precise time deltas.
+     */
+    public long absoluteTimeInNanos() {
+        return System.nanoTime();
+    }
+
```

Comment:
The name is misleading. Absolute time generally refers to wall clock time System#currentTimeMillis vs relative time is the logical time System#nanoTime

Reply:
I'm open to suggestions, how about `getNanoTime` or `getSystemNanoTime` instead of `absoluteTimeInNanos`? Don't want to use `relative` word here since then the concern will be relative to what.

Reply:
System.nanoTime() gives you the number of nanoseconds since an unknown and arbitrary point. Therefore it is "relative" in that it is only useful when you compare it relative to another invocation to System.nanoTime in the same JVM instance. So "relative" does make sense here, but the challenge is to differentiate it from the existing "relativeTime..." methods.

I might suggest something simple like "uncachedRelativeTimeInNanos" but I'd be happy to hear other suggests. Make sure the javadoc gives some guidance as to when to use this versus the other "relativeTime..." methods.

Reply:
@andrross @Bukhtawar Thanks for the suggestions. Putting more thoughts through, I'd prefer `getSystemNanoTime` to simply state that it's a wrapper over `System.nanoTime()`. Anything else in the method name will demand extra information to be conveyed through javadoc and method definition.

While javadoc should give guidance on "when" to use this method versus other relative time, the name should be clear and convey "what" it does without having to look through function definition / javadoc.

Thoughts?

Reply:
"what" it does is give you a precise relative time. System.nanoTime is "how" it does it, and the caller should not need to be concerned with the "how". What do you think of the following?

```
/**
 * Returns a value of nanoseconds that may be used for relative time calculations
 * that require the highest precision possible. Performance critical code must use
 * either {@link #relativeTimeInNanos()} or {@link #relativeTimeInMillis()} which
 * give better performance at the cost of lower precision.
 *
 * This method should only be used for calculating time deltas. For an epoch based
 * timestamp, see {@link #absoluteTimeInMillis()}.
 */
public long preciseRelativeTimeInNanos() {
    return System.nanoTime();
}
```

Reply:
@andrross I guess your latest suggestion makes sense. I guess thumbs up by @Bukhtawar & @shwetathareja shows their agreement as well. I've refactored accordingly.